lib/moneta/mixins.rb

```diff
@@ -233,6 +243,89 @@ def self.included(base)
     end
   end
 
+  # @api private
+  module EachKeySupport
+    def self.prepended(base)
+      base.supports(:each_key) if base.respond_to?(:supports)
+      require 'set'
+    end
+
+    # (see Defaults#each_key)
+    def each_key
+      return enum_for(:each_key) unless block_given?
+      @all_keys&.each{ |k| yield(k) }
```

Comment:
I think this method should return `self` when no block is given.

Reply:
Also, there should be a spec for returning `self`

Reply:
@asppsa why self? The point here is to be able to return an enumerator so each adapter is able to implement the iterator as is convinient, making use of lazy if needed.

I don't understand why I would like to return just self on a method where its "contract" is to return an iterator (or yield values to the block).
See Ruby's [Hash#each_key](https://ruby-doc.org/core-2.4.0/Hash.html#method-i-each_key)

Reply:
Oops!  You are right, I meant the opposite of what I said